### PR TITLE
[xtensa] Added add_platform_headers hook in CodeGen_C and relocated the common Xtensa code there.

### PR DESCRIPTION
**Pull request by Aelphy**

### CODE REVIEW

File: src/CodeGen_C.h [hunk 1]
```diff
@@ -115,6 +115,9 @@ class CodeGen_C : public IRPrinter {
     /** Emit a version of a string that is a valid identifier in C (. is replaced with _) */
     virtual std::string print_name(const std::string &);
 
+    /** Add platform speciefic headers. Needed for Xtensa */
```
Comment (vksnk) [inline src/CodeGen_C.h:118]: speciefic -> specific

Comment (vksnk) [inline src/CodeGen_C.h:118]: I don't think "Needed for Xtensa" part of the comment is needed, because this might be used on other platforms too.

**Review by vksnk — APPROVED** [review]

**Comment by steven-johnson** [comment]
Ready to land?

**Comment by vksnk** [comment]
This now has a merge conflict, it seems easy to resolve, but I don't think I can push into Misha's fork. Misha, could you please fix the conflict?

**Review by vksnk — APPROVED** [review]

**Pull request merged**